src/world/action/slide.c

```diff
@@ -1,5 +1,202 @@
+    sp48 = player_test_move_with_slipping(playerStatus, &sp28, &sp2C, &sp30, tempCurrentSpeed, playerStatus->heading);
+    playerStatus->position.x = sp28;
+    playerStatus->position.z = sp30;
+    playerStatus->position.y = sp2C;
+    switch (playerStatus->fallState) {
+        case 0:
+            D_802B6780 += 0.1;
+            if (D_802B6784 <= D_802B6780) {
+                D_802B6780 = D_802B6784;
+            }
+            playerStatus->currentSpeed += D_802B6780;
+            if (D_802B6788 <= playerStatus->currentSpeed) {
+                playerStatus->currentSpeed = D_802B6788;
+            }
+            sp28 = playerStatus->position.x;
+            sp34 = 100.0f;
+            sp30 = playerStatus->position.z;
+            D_802B6794 = D_802B6798;
+            sp2C = playerStatus->position.y + (playerStatus->colliderHeight * 0.5f);
+            sp48 = player_raycast_below_cam_relative(playerStatus, &sp28, &sp2C, &sp30, &sp34, &sp38, &sp3C, &sp40, &sp44);
+            D_802B6798 = sp3C;
+            if (sp48 >= 0) {
+                collisionStatus = &gCollisionStatus;
+                colliderTemp = get_collider_type_by_id(sp48);
+                if (colliderTemp == 5) {
+                    collisionStatus->currentFloor = sp48;
+                    playerStatus->position.y = sp2C;
+                    D_802B6790 = sp3C + 180.0f;
+                    break;
+                }
+                if (!(fabs(D_802B6794 - D_802B6798) >= 50.0)) {
+                    playerStatus->fallState = 1;
+                } else {
+                    playerStatus->fallState = 2;
+                }
+            } else {
+                playerStatus->fallState = 2;
+            }
+            sin_cos_rad((D_802B6790 * 6.28318f) / 360.0f, &sp4C, &sp50);
+            playerStatus->position.y += fabsf((sp4C / sp50) * playerStatus->currentSpeed);
+            snd_stop_sound(0x167);
+            break;
+        case 1:
+            sp28 = playerStatus->position.x;
+            sp34 = 50.0f;
+            sp30 = playerStatus->position.z;
+            sp2C = playerStatus->position.y + (playerStatus->colliderHeight * 0.5f);
+            sp48 = player_raycast_below_cam_relative(playerStatus, &sp28, &sp2C, &sp30, &sp34, &sp38, &sp3C, &sp40, &sp44);
+            if (sp48 >= 0) {
+                tempCurrentSpeed = playerStatus->currentSpeed / 3.0f;
+                if (tempCurrentSpeed < 0.01) {
+                    playerStatus->currentSpeed = 0.0f;
+                }
+                playerStatus->currentSpeed -= tempCurrentSpeed;
+                if (playerStatus->currentSpeed <= 0.0f) {
+                    sfx_play_sound_at_player(SOUND_UNKNOWN_172, 0);
+                    suggest_player_anim_setUnkFlag(0x10031);
+                    playerStatus->fallState = 6;
+                    playerStatus->framesOnGround = 15;
+                    playerStatus->currentSpeed = 0.0f;
+                    playerStatus->position.y = sp2C;
+                    break;
```

Comment:
This break in the if shouldn't be necessary since the next one will be hit immediately.